### ISSUE TITLE
Support GitHub enterprise in pluginDownloadUrl

### ISSUE THREAD

**Issue by Frassle** [pulumi/pulumi#11429]
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details

Github Enterprise API endpoints usually contain hierarchical slashes. An example format of a private host would be "github_hostname/api/v2" . This is not supported with the current method of splitting the PluginDownloadURL. We can't assume the last two parts are the Organization and Repo since the Repo is an optional field.

**Comment by rdanno** [comment]
Would be great to see this implemented!

**Comment by megbetts** [comment]
Yes please, we have to resorted to packaging the plugin with the sdk and pushing it to our pypi repo in the interim which isn't an ideal solution
